src/mem/ruby/system/Sequencer.cc

```diff
@@ -594,6 +600,57 @@ Sequencer::readCallback(Addr address, DataBlock& data,
     }
 }
 
+void
+Sequencer::atomicCallback(Addr address, DataBlock& data,
```

Comment:
I'm having trouble following the logic in this function. Could you explain a bit about what it's doing and why?

Reply:
Some operations can be coalesced with an Atomic Memory Operation. However, if the atomic was performed remotely, it is impossible to perform further atomics/writes in the same cache line. Therefore, we reissue the other requests to the ruby system.

We could make use of the already existing WriteCallback. However, we thought the proper way to address this new feature is to create a specific callback for atomic operations. This also enables future features like coalescing atomic operations as described in #200 but for CPU rather than GPU.

Reply:
I see. This makes sense to me now. However, I don't think the comments are doing a good job communicating the design. E.g., "perform hitCallback on first cpu request" is actually iterating through all requests and "free the ruby request form the list" is about 30 lines away from where that's actually done.

It would improve this change to update the comments in this new function.